### PR TITLE
Exception handler tests

### PR DESCRIPTION
**Pull request by Anish9901**
Related to #3524

This adds tests for the error handling decorator `handle_rpc_exceptions`, and verifies a function wrapped always raises a `RPCException`

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/tests/rpc/exceptions/test_error_codes.py [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-import pytest
+from django.core.exceptions import FieldDoesNotExist
+from mathesar.utils.connections import BadInstallationTarget
+from db.functions.exceptions import UnknownDBFunctionID
 from sqlalchemy.exc import IntegrityError
+from http.client import CannotSendRequest
 
-from db.functions.exceptions import UnknownDBFunctionID
-from mathesar.utils.connections import BadInstallationTarget
 import mathesar.rpc.exceptions.error_codes as err_codes
```
Comment (mathemancer) [inline mathesar/tests/rpc/exceptions/test_error_codes.py:10]: Did an automated tool reorder these imports? The new order is pretty non-standard. I'd rather have:

```python
from http.client import CannotSendRequest

from django.core.exceptions import FieldDoesNotExist
from psycopg.errors import BadCopyFileFormat
import pytest
from sqlalchemy.exc import IntegrityError

from db.functions.exceptions import UnknownDBFunctionID
import mathesar.rpc.exceptions.error_codes as err_codes
from mathesar.utils.connections import BadInstallationTarget
```

Comment (mathemancer) [reply mathesar/tests/rpc/exceptions/test_error_codes.py:10]: The imports should be grouped into:

1. standard library
2. 3rd party libraries
3. local libraries (our code)

Sorting should be lexicographical by non-keyword within those groups.

File: mathesar/tests/rpc/exceptions/test_error_handler.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import pytest
+
+from modernrpc.exceptions import RPCException
+from psycopg.errors import BadCopyFileFormat
+from django.core.exceptions import FieldDoesNotExist
+from mathesar.utils.connections import BadInstallationTarget
+from db.functions.exceptions import UnknownDBFunctionID
+from http.client import CannotSendRequest
+
+from mathesar.rpc.exceptions.handlers import handle_rpc_exceptions
```
Comment (mathemancer) [inline mathesar/tests/rpc/exceptions/test_error_handler.py:10]: These are also sorted strangely.

**Review by mathemancer — APPROVED** [review]
After discussing with @Anish9901 , I'm convinced there is a logic to the sorting of the imports in these test files.

@Anish9901 , Please add a comment or two about that logic, and then go ahead and merge this.

**Comment by Anish9901** [comment]
I've sorted & grouped the imports based on _"what's being tested"_ rather than _"where are we importing from"_. Furthermore, if we try to see the imports through the lens of Arrange, Act and Assert we'll notice that the following imports are required for the _Arrange_ step for the tests defined:
```py
from modernrpc.exceptions import RPCException
from psycopg.errors import BadCopyFileFormat
from django.core.exceptions import FieldDoesNotExist
from mathesar.utils.connections import BadInstallationTarget
from db.functions.exceptions import UnknownDBFunctionID
from http.client import CannotSendRequest
```
Similarly, we can think of the following import as something which will be _Acted_ upon or rather the function that is to be tested. 
```py
from mathesar.rpc.exceptions.handlers import handle_rpc_exceptions
```
The first line is reserved for the imports that would be imported multiple times throughout our test files and for any imports that are required for rudementry python/pytest functions e.g. `os`, `pytest` etc.

**Pull request merged**